This file is part of MXE. See LICENSE.md for licensing information.

The patch was taken from the scummvm.org documentation, it is
required on 64 bit systems due to the very old codebase.

--- a/libvo/video_out_dx.c.orig   2014-02-17 16:38:24.000000000 +0100
+++ b/libvo/video_out_dx.c        2014-02-17 16:39:34.000000000 +0100
@@ -92,7 +92,7 @@
     switch (message) {
 
     case WM_WINDOWPOSCHANGED:
-	instance = (dx_instance_t *) GetWindowLong (hwnd, GWL_USERDATA);
+	instance = (dx_instance_t *) GetWindowLongPtr (hwnd, GWLP_USERDATA);
 
        /* update the window position and size */
        point_window.x = 0;
@@ -173,7 +173,7 @@
     /* store a directx_instance pointer into the window local storage
      * (for later use in event_handler).
      * We need to use SetWindowLongPtr when it is available in mingw */
-    SetWindowLong (instance->window, GWL_USERDATA, (LONG) instance);
+    SetWindowLongPtr (instance->window, GWLP_USERDATA, (LONG_PTR) instance);
 
     ShowWindow (instance->window, SW_SHOW);
